src/core/thread/uri_paths.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file includes definitions for Thread URIs.
+ */
+
+#ifndef URI_PATHS_HPP_
+#define URI_PATHS_HPP_
+
+#include "openthread-core-config.h"
+
+namespace ot {
+
+/**
+ *
+ * This structure contains Thread URI Path string definitions.
+ *
+ */
+struct UriPath
+{
+    static const char kAddressQuery[];           ///< The URI Path for Address Query ("a/aq").
```

Comment:
By changing `struct UriPath` to `namespace UriPath`, I think we can defines the value directly at here:
```suggestion
    static const char kAddressQuery[] = "a/aq";
```

Reply:
Yes. This can work/compile (with or even without `static` since `const` definitions should have "internal linkage" model - I think after C++11), but I intentionally avoided this pattern:
- These array definitions are in a header file which is included by a bunch of `cpp` files.
- With this we will have the same const variable array definitions in each `cpp` file (scoped to that file).
- While most toolchain would optimize this while linking and will have a single common version of the array string, I don't think this is necessarily required or ensured (it's tool-chain dependent).
- The code in this PR is explicit about this and ensures that we will have a single instance of each array which is shared by all users.